mesonbuild/backend/vs2010backend.py

```diff
@@ -123,7 +148,17 @@ def generate_custom_generator_commands(self, target, parent_node):
     def generate(self, interp):
         self.resolve_source_conflicts()
         self.interpreter = interp
-        self.platform = 'Win32'
+        target_machine = self.interpreter.builtin['target_machine'].cpu_family_method(None, None)
+        if target_machine.endswith('64'):
+            # amd64 or x86_64
+            self.platform = 'x64'
+        elif target_machine == 'x86':
+            # x86
+            self.platform = 'Win32'
+        elif 'arm' in target_machine.lower():
+            self.platform = 'ARM'
+        else:
+            raise MesonException('Unsupported Visual Studio platform: ' + target_machine)
```

Comment:
I know I'm extremely late to the part here, but shouldn't this have consultet `host_machine` instead of `target_machine`? The docs indicate that `target_machine` is only for compiler-type projects, but I currently need to set it to get Visual Studio to cross-compile...